pylivetrader/backend/alpaca.py

```diff
@@ -495,48 +499,49 @@ def get_bars(self, assets, data_frequency, bar_count=500):
             symbols = [assets.symbol]
         else:
             symbols = [asset.symbol for asset in assets]
+        # the timezone of end_dt if UTC, when data_frequency=daily, end_dt is midnight in UTC time
+        if not end_dt:
+            end_dt = pd.to_datetime('now', utc=True).floor('min')
+            session_label = self._cal.minute_to_session_label(end_dt)
+            session_open = self._cal.session_open(session_label)
+            session_close = self._cal.session_close(session_label)
+            if end_dt < session_open:  # before market open
+                end_dt = self._cal.previous_minute(end_dt)
+            elif session_open <= end_dt < session_close:
+                pass
+            else:
+                end_dt = session_close
+            if is_daily:
```

Comment:
check this before you do the entire if block above. it is not needed if you are using daily bars